src/prefect/core/task.py

```diff
@@ -267,11 +267,11 @@ def __init__(
 
         self.target = target
 
-        if result.location and target:
+        if hasattr(result, "location") and target:
```

Comment:
I think you mean `getattr(result, "location", None)`

Reply:
Yup I did